### ISSUE TITLE
Empty TypeList blocks are stripped in 0.12

### ISSUE THREAD

**Issue by rileykarson** [hashicorp/terraform#19760]
### Terraform Version
```
Terraform 0.12alpha4 vendored into the Google provider (https://github.com/terraform-providers/terraform-provider-google/pull/2753)
```

### Terraform Configuration Files
```hcl
resource "google_compute_instance" "foobar" {
	name           = "my-instance"
	machine_type   = "n1-standard-1"

	boot_disk {
		initialize_params{
			image = "${data.google_compute_image.my_image.self_link}"
		}
	}

	network_interface {
		network = "default"
		access_config { }
	}
}
```
The specific block here is `network_interface.access_config`

Source: https://github.com/terraform-providers/terraform-provider-google/blob/master/google/resource_compute_instance_test.go#L1771

### Expected Behavior
Terraform should send an array of `nil`, or (preferably) an array containing a single empty `map[string]interface` to the provider

### Actual Behavior
Terraform omits the field entirely 

### Steps to Reproduce
```
make testacc TEST=./google TESTARGS='-parallel=1 -run=TestAccComputeInstance_update'
```

on https://github.com/terraform-providers/terraform-provider-google/pull/2753

### Additional Context
Running in CI + locally as a test, I'm a Google provider maintainer, we've vendored `0.12alpha4` in our branch

`access_config` is "load bearing" even when empty; omitting the field entirely is the  user stating they don't want their VM instance to be accessible, while including an empty definition means it will take on API-defined default values such as the IP address.

**Comment by jbardin** [comment]
Hi @rileykarson,

Unfortunately, relying on an empty value isn't something that is fully supported by helper/schema., because there really is no way to differentiate between empty and unset. While this may work in some cases while setting a value, there would be no way to unset the value to an empty list. The new plugin API does support the ability to differentiate between unset, and empty values, but that's not something we can access until we get a new plugin SDK. 

I'll see if I there is any way to get the schema to detect the initial empty value, but we may want to reconsider the way this is configured in order to prevent the issues associated with users attempting to update the value. (IIRC there are a number of other google provider issues around this)

**Comment by danawillow** [comment]
> The new plugin API does support the ability to differentiate between unset, and empty values, but that's not something we can access until we get a new plugin SDK.
What's the difference between that and the new provider sdk that we vendored in? In the 0.11 series, we were able to tell the difference between empty and unset for objects like this, because empty would return us a list of size one with an empty map in it, whereas unset would return a list of size zero. 

And yeah this is the only instance I know of in the google provider that relies on a nested object being set but empty, but I can't speak to any of the non-google providers.

If it's something we have to change in the provider, then I'd say we add a top-level boolean called `ephemeral_ip` or something like that, but I agree that I'd prefer that we don't have to change our behavior if possible.

**Comment by jbardin** [comment]
Thanks, I may be thinking of issues from other providers, but this is not an unknown situation (e.g. most of the issues relating to `GetOK` or `GetOKExists`). This is yet a little different, because it's not just relying on an empty value, but an empty block which there was no real representation of, and as mentioned above requires accessing the implementation detail of seeing a single empty `map[string]interface{}`. 

While I'd like to fix the current shims to emulate the legacy behavior as closely as possible, the fundamental problem is `helper/schema` has no way to distinguish between unset and empty values, and relying on empty values is only likely to cause other bugs.

**Comment by rileykarson** [comment]
I'm still experiencing this at f65b7c5; would you mind clarifying what we expect to change after the fix? Based on communication with @paddycarver we'd expected that there would be no changes that affect us.

**Comment by jbardin** [comment]
I've confirmed this test passes on master.
I'm going to wait until I get a fix in for #20082 and double check before closing, to make sure that my reproduction of this issue is still equivalent.

**Comment by jbardin** [comment]
Confirmed fixed in master
